### PR TITLE
Adds regular branch protection checks

### PR DESCRIPTION
**Pull request by ebenoist**
On the regular cron, ensure that the project's branch protection configuration is correct.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ lint: ## check for formatting errors
 
 .PHONY: test
 test: ## verify functionality
-	jest
+	GITHUB_TOKEN='foozles' jest
```
Comment (stevenweber) [inline Makefile:36]: What is this for?

File: __tests__/check_for_branch_protection.test.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        required_pull_request_reviews: {
+          dismiss_stale_reviews: true,
+        },
+        required_status_checks: {
+          contexts: [{
+            name: 'ci',
+          }],
+        }
+      },
```
Comment (stevenweber) [inline __tests__/check_for_branch_protection.test.ts:48]: #nit would be easier to see the relevant part if the ok data was extracted to a constant, e.g.

```js
const VALID_CONFIG = {
  //...
}

(client.repos.getBranchProtection as any).mockReturnValue(Promise.resolve({
  ...VALID_CONFIG,
  enforce_admins: {
    enabled: false,
  },
}))
```

File: src/input.ts [hunk 3]
```diff
@@ -22,5 +23,6 @@ export function getInput(): Input {
     slackHook: core.getInput('slack_hook'),
     posthocApprovalLabel: core.getInput('posthoc_approval_label') || 'posthoc-approval',
     verifiedCILabel: core.getInput('verified_ci_label') || 'verified-ci',
+    defaultBranch: core.getInput('default_branch') || 'master',
```
Comment (stevenweber) [inline src/input.ts:26]: Why not use github action defaults now that they're working again, i.e. just have an input named branch that defaults to 'master'?

File: Makefile (hunk 1 above)
Comment (ebenoist) [reply Makefile:36]: The github client will use the GITHUB_TOKEN env var if present. This prevents it from being used in test.

File: __tests__/check_for_branch_protection.test.ts (hunk 2 above)
Comment (ebenoist) [reply __tests__/check_for_branch_protection.test.ts:48]: Probably? I'm not a huge fan of DRY tests, but I think thats probably a good middleground.

File: src/input.ts (hunk 3 above)
Comment (ebenoist) [reply src/input.ts:26]: I didn't realize they are working again! Happy to lean into those. The default iirc is `main` for all new repos, but `master` for anything made before that change.

File: Makefile (hunk 1 above)
Comment (stevenweber) [reply Makefile:36]: Wooza, scary it doesn't check an env flag with it.

**Review by stevenweber — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (ebenoist) [reply Makefile:36]: Yeah. This is how I figured out why my tests were working locally and not in ci :/

**Pull request merged**